source/workflowapi/app.py

```diff
@@ -1952,8 +1952,8 @@ def create_workflow_execution(trigger, workflow_execution):
                 asset_input = {
                     "Media": {
                         media_type: {
-                            "S3Bucket": asset_creation["S3Bucket"],
-                            "S3Key": asset_creation["S3Key"]
+                            "S3Bucket": s3bucket,
+                            "S3Key": s3key
```

Comment:
double check that this is necessary. Maybe the user-defined input is already in the asset_creation object.
Also think about how we're going to fail on reprocess. If the source is gone, then log a failure that the source is gone.